extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/VertxHttpRecorder.java

```diff
@@ -215,6 +219,36 @@ public void handle(RoutingContext routingContext) {
             });
         }
 
+        if (httpConfiguration.limits.maxBodySize.isPresent()) {
+            long limit = httpConfiguration.limits.maxBodySize.get().asLongValue();
+            Long limitObj = limit;
+            router.route().order(-2).handler(new Handler<RoutingContext>() {
+                @Override
+                public void handle(RoutingContext event) {
```

Comment:
Any reason to not use https://vertx.io/docs/apidocs/io/vertx/ext/web/handler/BodyHandler.html#setBodyLimit-long-

Reply:
We do, but not everything uses the body handler, as it requires the full request to be buffered.

Reply:
oh I see! Thanks!